### PR TITLE
Fixes #6899 - Remove name_sort field

### LINKED ISSUE #6899: Remove name_sort field
We implemented `name.raw` in #6895 so now `name_sort` is not needed anymore.

* [ ] Change default field that is being used in `SortingFilter` to `name.raw`
* [ ] Remove field from addons.indexers

### PR DESCRIPTION
**Pull request by SeanPrashad**
Fixes #6899:

- [x] Changed the default field that is being used in `SortingFilter` to `name.raw`
- [x] Removed `name_sort` from `addons.indexers`

### CODE REVIEW

File: src/olympia/addons/indexers.py [hunk 1]
```diff
@@ -152,19 +152,6 @@ def get_mapping(cls):
...
-                        'type': 'object',
-                        'properties': {
-                            'accentcolor': {'type': 'keyword', 'index': False},
-                            'author': {'type': 'keyword', 'index': False},
-                            'header': {'type': 'keyword', 'index': False},
-                            'footer': {'type': 'keyword', 'index': False},
-                            'is_new': {'type': 'boolean', 'index': False},
-                            'textcolor': {'type': 'keyword', 'index': False},
-                        }
-                    },
```
Comment (EnTeQuAk) [inline src/olympia/addons/indexers.py]: I think you removed too much, the issue was just about removing `name_sort`, the `persona` property needs to stay :)

Comment (SeanPrashad) [reply src/olympia/addons/indexers.py]: Thanks for that heads up! Just gotta figure out what tests are failing for Travis now

**Comment by diox** [comment]
Hey @SeanPrashad, just wanted to say I've watched the video you posted about your thought process regarding this issue and it was very cool, keep up the good work! You're close to finishing this PR, only [a couple test failures left that were found by travis](https://travis-ci.org/mozilla/addons-server/builds/341986977).

While you're at it, could you please remove `'*_sort'` from `AddonIndexer`'s `hidden_fields` in `src/olympia/addons/indexers.py` ? The purpose of that variable is to list the fields we don't need to fetch from ElasticSearch when making a search query (making the result smaller in size, improving performance), and once `name_sort` is gone we should not have any other fields that match this `'*_sort'` rule, so we no longer need it.

**Comment by SeanPrashad** [comment]
Thank you so much @diox - It definitely means a lot 😄 ! I've gotten a tiny bit closer tonight with Travis only complaining about a [different assertion](https://travis-ci.org/mozilla/addons-server/jobs/342175558#L1505). This one is a bit harder for me to interpret but I think it's not receiving the correct value in both arrays because sorting is not done - what do you think?

**Comment by diox** [comment]
That test indicates that there is more work to be done :) Looks like it's legitimately broken because the searching for addons and sorting by name is broken in that particular area of the code. There are 3 more occurrences of `name_sort` in `src/olympia/search/views.py`, and 2 of them need to be changed: the one in `_personas()` and the one in `search()`.

(The reason the one in `_collections()` should be not changed in that searching for collections uses a different indexer, in `src/olympia/bandwagon/indexers.py`, and that one was not updated to switch to `name.raw` - it's still using `name_sort` only. Not worth fixing right now)

**Comment by SeanPrashad** [comment]
Thank you @diox for those final tidbits of guidance! It looks like TravisCI is giving us the green light to merge - did you want me to squash any of my commits together or is it fine with how it currently is?

**Comment by diox** [comment]
That looks good, thanks. I'll just squash it through github, no need to rebase it yourself.

**Pull request merged**